doc/doxy/Doxyfile

```diff
@@ -1,187 +1,919 @@
-# Doxyfile 1.5.5
+# Doxyfile 1.8.17
+
+# This file describes the settings to be used by the documentation system
+# doxygen (www.doxygen.org) for a project.
+#
+# All text after a double hash (##) is considered a comment and is placed in
+# front of the TAG it is preceding.
+#
+# All text after a single hash (#) is considered a comment and will be ignored.
+# The format is:
+# TAG = value [value, ...]
+# For lists, items can also be appended using:
+# TAG += value [value, ...]
+# Values that contain spaces should be placed between quotes (\" \").
```

Comment:
There are many additional comments in this file. Is this intentional?

Reply:
This is automatic, this file is generated after upgrading to Doxyfile version 1.8.17.
The additions are almost all comments (probably manually removed from previous versions ?) but there are some extra options added and some old ones removed.

Reply:
Do we really need these additional comments? The file is long enought without them.

Reply:
We do not need them but could be useful as a documentation. Also to my understanding one has to remove extra comments manually after every Doxyfile upgrade and that could be error prone. But I am OK removing those if you insist because I guess a similar decision had already been taken in the past.

Reply:
> remove extra comments manually after every Doxyfile upgrade

I thought you made this on purpose. AFAIU `doxygen` can generate a warning if something is wrong with the input file but will not modify it unless you pass `-u` option. This option is not passed in `make_qbk.py` so you must have passed it by yourself.

For me doxygen 1.9.1 reports:
```
warning: Tag 'TCL_SUBST' at line 39 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'COLS_IN_ALPHA_INDEX' at line 172 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'PERL_PATH' at line 295 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'MSCGEN_PATH' at line 300 of file 'Doxyfile' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
```

So I'd simply remove these variables from the file if I wanted to silence this warning or leave the doxyfile unmodified, as it is now. Because this is not related to `closest_points` in any way.